chapters/modules/chapter_8/pages/sharp.adoc

```diff
@@ -0,0 +1,67 @@
+
+The main advantage of the SHARP system is its ability to reduce costs and improve efficiency in the StarkNet network. By aggregating multiple Cairo jobs, the protocol can take advantage of the exponential acceleration provided by the STARK proof system. As a result, each individual transaction costs much less, making the entire process more affordable for users.
+
+Additionally, the SHARP system allows smaller users with limited computation to benefit from joining other jobs and sharing the cost of generating the proofs. This collaborative approach is similar to using public transportation instead of a private car, where the cost is distributed among all participants, making it more affordable for everyone.
+
+== Recursive Proofs in the SHARP System
+
+One of the most interesting aspects of the SHARP system is its use of recursive proofs. Instead of sending the generated proofs directly to the Solidity verifier, the proofs are first sent to a Cairo verifier, which is a STARK verifier written in Cairo. The Cairo verifier then checks the proofs off-chain and generates a new proof, which can be sent back to the SHARP system.
+
+This process continues in a recursive manner, with each new proof being sent to the Cairo verifier until a trigger is reached. At this point, the last proof in the series is sent to the Solidity verifier on Ethereum. This approach allows for greater parallelization of the computation and reduces the time and cost associated with generating and verifying proofs.
+
+At first glance, recursive proofs may seem more complex and time-consuming. However, there are several benefits to this approach:
+
+. *Parallelization*: Recursive proofs allow for parallelization of work, reducing latency for users and improving system efficiency.
+. *Lower cloud costs*: Since each job is shorter, the required memory for processing is reduced, resulting in lower cloud costs.
+. *Optimization*: Recursive proofs enable the system to optimize for various factors, including latency, on-chain costs, and time to proof.
+
+== SHARP Backend Architecture and Data Pipeline
+
+The SHARP backend architecture consists of several services that work together to process Cairo jobs and generate proofs. These services include the gateway, validator, scheduler, Cairo runner, prover, and dispatcher.
+
+. *Gateway*: Cairo jobs enter the system through the gateway.
+. *Validator*: The validator service runs validation checks on each job, ensuring they meet the necessary requirements and can fit within the prover machines. Invalid jobs are tagged as such and do not proceed to the prover.
+. *Scheduler*: The scheduler service is responsible for creating "trains" that aggregate jobs and send them to the prover. Recursive jobs are paired and sent to the prover together.
+. *Cairo Runner*: This service runs Cairo for the prover's needs.
```

Comment:
Is the execution trace generated at that step or the folllowing?

Reply:
To the best of my knowledge, coming from the call, it is generated at this step. 

I added the following text:

```
. *Cairo Runner*: This service runs Cairo for the prover's needs. That is, the Cairo Runner service runs Cairo programs, executing the necessary computations, and generating the execution trace as an intermediate result. This execution trace is then used by the Prover.
```